contributors/design-proposals/raw-block-pv.md

```diff
@@ -0,0 +1,669 @@
+  implemented in the community. In addition, in-tree provisioners should be able to gracefully ignore volumeMode API objects for plugins
+  that haven't been updated to accept this value.
+  
+  It is important to note that when a PV is bound, it is either bound as a raw block device or formatted with a filesystem. Therefore, 
+  the PVC drives the request and intended usage of the device by specifying the volumeMode as part of the API. This design lends itself
+  to future support of dynamic provisioning by also letting the request initiate from the PVC defining the role for the PV. It also 
+  allows flexibility in the implementation and storage plugins to determine their support of this feature. Acceptable values for 
+  volumeMode are 'Block' and 'Filesystem'. Where 'Filesystem' is the default value today and not required to be set in the PV/PVC.
+  
+# Proposed API Changes
+   
+## Persistent Volume Claim API Changes:
+In the simplest case of static provisioning, a user asks for a volumeMode of block. The binder will only bind to a PV defined 
+with the same volumeMode.
+
+```
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+name: myclaim
+spec:
+  volumeMode: Block #proposed API change
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 80Gi 
+```
+
+For dynamic provisioning and the use of the storageClass, the admin also specifically defines the intent of the volume by 
+indicating the volumeMode as block. The provisioner for this class will validate whether or not it supports block and return
+an error if it does not.
+
+```
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+name: myclaim
+spec:
+  storageClassName: local-fast 
+  volumeMode: Block #proposed API change
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 80Gi 
+```
+
+## Persistent Volume API Changes:
+For static provisioning the admin creates the volume and also is intentional about how the volume should be consumed. For backwards
+compatibility, the absence of volumeMode will default to filesystem which is how volumes work today, which are formatted with a filesystem depending on the plug-in chosen. Recycling will not be a supported reclaim policy as it has been deprecated. The path value in the local PV definition would be overloaded to define the path of the raw block device rather than the fileystem path.
+```
+kind: PersistentVolume
+apiVersion: v1
+metadata:
+name: local-raw-pv
+spec:
+  volumeMode: Block #proposed API change
+  capacity:
+    storage: 100Gi
+  local:
+    path: /dev/xvdc #device path
+  accessModes:
+    - ReadWriteOnce
+  persistentVolumeReclaimPolicy: Delete 
+```
+## Pod API Changes:
+This change intentionally calls out the use of a block device (volumeDevices) rather than the mount point on a filesystem.
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: my-db
+spec:
+    containers
```

Comment:
s/containers/containers:  (add the colon) or you will get a json/yaml validation error

```
   ...
    containers:
    - name: mysql
    ...
```

Reply:
@screeley44 done